openfecwebapp/views.py

```diff
@@ -33,32 +22,65 @@ def render_table(data_type, results, params):
     results_table = {}
     results_table[data_type] = []
 
-    results_table['pagination'] = generate_pagination_values(
-        results, params, data_type)
-
-    for r in results['results']:
-        results_table[data_type].append(type_map[data_type](r))
+    results_table['pagination'] = generate_pagination_values(results, params, data_type)
+    results_table[data_type] = results['results']
 
     if params.get('name'):
         results_table['filter_name'] = params['name']
 
     return render_template(data_type + '.html', **results_table)
 
-type_map = {
-    'candidates': lambda x: x,
-    'candidate': map_candidate_page_values,
-    'committees': lambda x: x,
-    'committee': map_committee_page_values
-}
 
-def render_page(data_type, c_data):
-    # not handling error at api module because sometimes its ok to 
-    # not get data back - like with search results
-    if not 'results' in c_data or not c_data['results']:
-        abort(500)
+def render_committee(data, candidates=None):
+    committee = get_first_result_or_raise_500(data)
+
+    # committee fields will be top-level in the template
+    tmpl_vars = committee
+
+    # add related candidates a level below
+    tmpl_vars['candidates'] = candidates
+
+    financials = load_cmte_financials(committee['committee_id'])
+    tmpl_vars['reports'] = financials['reports']
+    tmpl_vars['totals'] = financials['totals']
+
+    return render_template('committees-single.html', **tmpl_vars)
 
-    tmpl_vars = c_data['results'][0]
-    tmpl_vars.update(type_map[data_type](c_data['results'][0]))
-    tmpl_vars.update(add_cmte_financial_data(tmpl_vars, data_type))
 
-    return render_template(data_type + 's-single.html', **tmpl_vars)
+def render_candidate(data, committees=None):
+    results = get_first_result_or_raise_500(data)
+
+    # candidate fields will be top-level in the template
+    tmpl_vars = results
+    tmpl_vars['election_years'] = load_election_years(results['candidate_id'])
+
+    # add 'committees' level to template
+    tmpl_vars['has_authorized_cmtes'] = False
+    tmpl_vars['has_joint_cmtes'] = False
+    tmpl_vars['has_leadership_cmtes'] = False
+
+    for committee in committees:
+        if committee['designation'] in ('P', 'A'):  # (P)rimary or (A)uthorized
+            # this adds committee['reports'] and committee['totals']
+            committee.update(load_cmte_financials(committee['committee_id']))
+            tmpl_vars['has_authorized_cmtes'] = True
+
+        elif committee['designation'] == 'J':  # (J)oint
+            tmpl_vars['has_joint_cmtes'] = True
+
+        elif committee['designation'] == 'D':  # Leadership
+            tmpl_vars['has_leadership_cmtes'] = True
+
+    tmpl_vars['committees'] = committees
+    from marshmallow import pprint; pprint(tmpl_vars)
+
+    return render_template('candidates-single.html', **tmpl_vars)
+
+
+def get_first_result_or_raise_500(data):
```

Comment:
This behavior predates this patch, so we can defer discussion until later, but I'm not sure we should be throwing a 500 here. This behavior doesn't give much feedback to the user or to the developer, and it can potentially give incorrect feedback: for example, browsing to `/candidate/fakecandidate` should throw a 404, but this logic will throw a 500 instead.